src/libutil/util.cc

```diff
@@ -112,6 +112,21 @@ std::string rewriteStrings(std::string s, const StringMap & rewrites)
 }
 
 
+std::string renderSize(uint64_t value, bool align)
+{
+    static const std::array<char, 9> prefixes{{
+        'K', 'K', 'M', 'G', 'T', 'P', 'E', 'Z', 'Y'
+    }};
+    size_t power = 0;
+    double res = value;
+    while (res > 1024 && power < prefixes.size()) {
+        ++power;
+        res /= 1024;
+    }
+    return fmt(align ? "%6.1f %ciB" : "%.1f %ciB", power == 0 ? res / 1024 : res, prefixes.at(power));
+}
+
+
```

Comment:
I see that this was *mostly* just moved, but I'm a bit concerned about the lack of a unit test for this. I'd be interested to see how some more examples look, and to see that they're correct.

Reply:
Added some tests.